test/integration/main_test.go

```diff
@@ -0,0 +1,505 @@
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		// Set up global resource manager client.
+		pc, err := resourcemanager.NewProjectsClient(ctx)
+		if err != nil {
+			log.Printf("failed to create projects client: %v", err)
+			return 2
+		}
+		defer pc.Close()
+		projectClient = pc
+
+		return m.Run()
+	}())
+}
+
+func TestIAMHandle(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name                 string
+		filePath             string
+		verbose              bool
+		conditionTitlePrefix string
+		wantBindings         []*iampb.Binding
+		wantOutput           string
+		wantErrSubstr        string
+	}{
+		{
+			name:     "success",
+			filePath: "iam.yaml",
+			wantBindings: []*iampb.Binding{
+				{
+					Role:    "roles/actions.Viewer",
+					Members: []string{"user:aod-test-bot@tycho.joonix.net"},
+					Condition: &expr.Expr{
+						Title:      cfg.ConditionTitle,
+						Expression: fmt.Sprintf("request.time < timestamp('%s')", cfg.IAMExpiration),
+					},
+				},
+				{
+					Role:    "roles/ml.viewer",
+					Members: []string{"user:aod-test-bot@tycho.joonix.net"},
+					Condition: &expr.Expr{
+						Title:      cfg.ConditionTitle,
+						Expression: fmt.Sprintf("request.time < timestamp('%s')", cfg.IAMExpiration),
+					},
+				},
+			},
+			wantOutput: fmt.Sprintf(`------Successfully Handled IAM Request------
+iamrequest:
+  policies:
+    - resource: projects/access-on-demand-i-12af76
+      bindings:
+        - members:
+            - user:aod-test-bot@tycho.joonix.net
+          role: roles/actions.Viewer
+        - members:
+            - user:aod-test-bot@tycho.joonix.net
+          role: roles/ml.viewer
+duration: %sh0m0s
+starttime: %s
+`, cfg.IAMExpirationDurationHour, cfg.IAMExpirationStartTime),
+		},
+		{
+			name:                 "invalid_file",
+			filePath:             "iam_invalid.yaml",
+			wantErrSubstr:        "failed to validate",
+			conditionTitlePrefix: "invalid-",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			fullTitle := fmt.Sprintf("%s%s", tc.conditionTitlePrefix, cfg.ConditionTitle)
+
+			args := []string{
+				"iam", "handle",
+				"-path", tc.filePath,
+				"-start-time", cfg.IAMExpirationStartTime,
+				"-duration", fmt.Sprintf("%sh", cfg.IAMExpirationDurationHour),
+				"-custom-condition-title", fullTitle,
+			}
+			if tc.verbose {
+				args = append(args, "-verbose")
+			}
+
+			stderr := bytes.NewBuffer(nil)
+			stdout := bytes.NewBuffer(nil)
+			cmd := exec.Command("aod", args...)
+			cmd.Dir = cfg.WorkingDir
+			cmd.Stdout = stdout
+			cmd.Stderr = stderr
+
+			// Sometimes the cmd exits with non-zero status when the command is run
+			// successfully but has error.
+			if err := cmd.Run(); err != nil {
```

Comment:
We don't get much value running binary, do we? It would require us to build the binary in test. I think we can just run this instead https://github.com/abcxyz/access-on-demand/blob/main/pkg/cli/root.go#L67C6-L67C45

Reply:
One big benefit is it includes main.go?

Reply:
It's not a _big_ benefit. What's in main.go?

Reply:
My concern is otherwise how to know if the build will be successful, but since we hardly change main.go, it should be fine to skip.